src/IonTextWriter.ts

```diff
@@ -216,8 +218,8 @@ export class TextWriter extends AbstractWriter {
         writer.writeUtf8(text);
     }
 
-    writeInt(value: number) : void {
-        this._serializeValue(IonTypes.INT, value, (value: number) => {
+    writeInt(value: number | JSBI) : void {
+        this._serializeValue(IonTypes.INT, value, (value: number | JSBI) => {
```

Comment:
This can be `null` right?

Reply:
It can. By default, Typescript treats `null` and `undefined` as [subtypes of all other types](https://www.typescriptlang.org/docs/handbook/basic-types.html#null-and-undefined), meaning they're allowed even if the type signature of a method doesn't accept it explicitly.

While the `Writer` interface's methods spell out where `null` is acceptable, the implementations don't for some reason.

You can opt into package-wide type checking that requires methods to explicitly allow `null`s via the `--strictNullChecks` flag. I'd be for doing this, but it would be a sweeping change better suited to its own PR.